### PR TITLE
Officially add SageMaker Notebooks support

### PR DESCRIPTION
**Pull request by benjaminpkane**
Adds SageMaker support with new `amzn` distribution support in `fiftyone-db` and improved proxy handling. Updates documentation FAQ, troubleshooting, and environments pages

### Testing

In SageMaker, [create a notebook instance](https://docs.aws.amazon.com/sagemaker/latest/dg/gs-setup-working-env.html)

Open the notebook and install the RCs with
```py
!RELEASE_VERSION=0.5.0rc1 pip install fiftyone-db==0.5.0rc1 fiftyone==0.23.0rc2
```

Launch the App with a configured proxy
```py
import fiftyone as fo
fo.app_config.proxy_url = "https://<notebook-instance>.notebook.us-east-2.sagemaker.aws/proxy/5151"

dataset, session = fo.quickstart()
```

The changes can also be tested in Colab, but only `session.url` is functional. See #3644 

```py
!RELEASE_VERSION=0.5.0rc1 pip install fiftyone-db==0.5.0rc1 fiftyone==0.23.0rc2
import fiftyone as fo

# in colab, proxy_url is not needed
dataset, session = fo.quickstart()
```
Screenshot 2023-11-06 at 4 05 28 PM

### CODE REVIEW

**Comment by Stijnp** [comment]
Very excited about this feature, tried it out on my end but my session is timing out: 

Pixelating... for a few minutes until it throws Error: 599/500 as per image below.

I tried in the following notebook instance environments: 
- running on Amazon Linux 2 (5.10.192-183.736.amzn2.x86_64)
- Jupyter Lab 1 and 3
- not running on a VPC
- root access enabled
- running in zone ap-southeast-2 and us-east-2
- Firefox and Chrome

Please let me know if I can do anything to help further testing this feature.

Screenshot 2023-11-13 at 11 31 40 am
Screenshot 2023-11-13 at 11 31 59 am

**Comment by benjaminpkane** [comment]
Hi @Stijnp! My mistake, `0.23.0rc2` should be used. I updated the description.

File: fiftyone/server/mutation.py [hunk 1]
```diff
@@ -91,7 +91,9 @@ async def set_dataset(
         state.color_scheme = build_color_scheme(
             None, state.dataset, state.config
         )
-        state.group_slice = state.dataset.group_slice
+        if state.dataset is not None:
```
Comment (manivoxel51) [inline fiftyone/server/mutation.py:94]: 👍

**Review by manivoxel51 — APPROVED** [review]
Nice work! 🍨

Created an instance and tested the app's basics on Sagemaker's jupyterLab . looking good!

**Pull request merged**